### ISSUE TITLE
Peak picking some MS levels only

### ISSUE THREAD

**Issue by lgatto** [compomics/ThermoRawFileParser#111]
Thank you for `ThermoRawFileParser`, which I happily use on Linux. 

It seems that peak picking is either activated on all scans or none, using `-p`. Unless I have missed it, it would be great to be able to apply peak picking only on some, use-defined MS levels. My typical use case would be to do it on MS2 scans and leave the MS1 scans in profile. I am using the latest release 1.3.3.

**Comment by caetera** [comment]
Hi @lgatto,
That is correct, it is either all centroided (default) or all profile, if it is present ( `-p` ). The feature you request can be implemented in the next release.

**Comment by lgatto** [comment]
That would be great, thank you. 

Is there a release data for the next one, or is it whenever it is ready?

**Comment by caetera** [comment]
No, we do not have the release date yet, but if you comfortable using prerelease build and/or build TRFP from the source, you can get it faster.

**Comment by lgatto** [comment]
Thanks, will do.

**Comment by lgatto** [comment]
Are there any build instructions available somewhere?

**Comment by caetera** [comment]
I don't think we have official building instructions, but in principle, it is just cloning the repository and building with `msbuild`, assuming you have complete mono installation. For older versions of mono `xbuild` should be used instead.

Below is the excerpt from the Dockerfile I use locally, the compiled version will be in `/home/biodocker/build/bin/x64/Debug/`
```
    git clone -b master --single-branch --depth 1 https://github.com/compomics/ThermoRawFileParser /home/biodocker/build 
    msbuild /home/biodocker/build
```

There is a Dockerfile included in the repository, it should work as well.

**Comment by lgatto** [comment]
It ran

```
git clone -b master --single-branch --depth 1 https://github.com/compomics/ThermoRawFileParser
cd ThermoRawFileParser
msbuild
```

Which seemed successful. But
```
$ mono bin/x64/Debug/ThermoRawFileParser.exe --version
1.3.3
```
still indicates version 1.3.3 and I can't spot anything in the documentation about picking peaks in some MS levels.

**Comment by caetera** [comment]
The feature is still to be implemented. I will update the issue when it will be ready. Sorry for the confusion.

**Comment by lgatto** [comment]
Ah ok, thanks.

**Comment by caetera** [comment]
273ff2cfaf6ad1b8e270e87a111fa23b94046c17 implements the feature
The `-p|--noPeakPicking` now supports additional value argument limiting it to some MS levels, the argument can be a list of numbers or intervals (including open) separated by commas, for example
just `-p` (no value defined) - no peak picking for all MS levels
`-p=2-` - no peak picking for MS2 and higher
`-p=1,3,5` - no peak picking for level MS1, MS3, and MS5
`-p=1` - no peak picking for MS1 only

Please, note, that due to the limitation of the command line parsing library optional arguments has to be defined without white spaces, i.e. any of `-p=1,2,3`, `-p:1,2,3`, `-p1,2,3` will work, but `-p 1,2,3` won't.

You can build it from branch [1.4.0](https://github.com/compomics/ThermoRawFileParser/tree/1.4.0), i.e.
`git clone -b 1.4.0 --single-branch --depth 1 https://github.com/compomics/ThermoRawFileParser`

**Comment by lgatto** [comment]
Yes, I confirm it works. Thanks!

**Comment by nielshulstaert** [comment]
This feature is available in version 1.3.4
